### ISSUE TITLE
Server: Define new "sort" field to Subject and SubSubject tables

### ISSUE THREAD

**Issue by TiE23** [TiE23/metric-teacher#222]
In #221 I realized that if I wasn't in a state where I was willing to completely obliterate the database and rebuild again from scratch that adding a new subsubject and sorting it on any page would become a complete nightmare.

**Should do after #221, by the way...**

So, I want to add a sort field (perhaps named `sort` or `serial` or `index`) to Subjects and SubSubject tables.

The SubSubjects table:
```
1001000 - Length - Hand-scale - To Metric
1001001 - Length - Hand-scale - From Metric
1002000 - Length - Human-scale - To Metric
1002001 - Length - Human-scale - From Metric
1003000 - Length - Room-scale - To Metric
1003001 - Length - Room-scale - From Metric
1004000 - Length - Building-scale - To Metric
1004001 - Length - Building-scale - From Metric
1005000 - Length - Local-scale - To Metric
1005001 - Length - Local-scale - From Metric
1006000 - Length - Geographic-scale - To Metric
1006001 - Length - Geographic-scale - From Metric
1007000 - Length - Astronomical-scale - To Metric
1007001 - Length - Astronomical-scale - From Metric
2000500 - Mass - Micro-scale - To Metric
2000501 - Mass - Micro-scale - From Metric
2001000 - Mass - Hand-scale - To Metric
2001001 - Mass - Hand-scale - From Metric
2002000 - Mass - Human-scale - To Metric
2002001 - Mass - Human-scale - From Metric
2003000 - Mass - Street-scale - To Metric
2003001 - Mass - Street-scale - From Metric
2004000 - Mass - Building-scale - To Metric
2004001 - Mass - Building-scale - From Metric
3001000 - Volume - Hand-scale - To Metric
3001001 - Volume - Hand-scale - From Metric
3002000 - Volume - Room-scale - To Metric
3002001 - Volume - Room-scale - From Metric
3003000 - Volume - Building-scale - To Metric
3003001 - Volume - Building-scale - From Metric
4001000 - Temperature - Weather-scale - To Metric
4001001 - Temperature - Weather-scale - From Metric
4002000 - Temperature - Cooking-scale - To Metric
4002001 - Temperature - Cooking-scale - From Metric
4003000 - Temperature - Chemistry-scale - To Metric
4003001 - Temperature - Chemistry-scale - From Metric
5001000 - Velocity - Pedestrian-scale - To Metric
5001001 - Velocity - Pedestrian-scale - From Metric
5002000 - Velocity - Transit-scale - To Metric
5002001 - Velocity - Transit-scale - From Metric
5003000 - Velocity - Racing-scale - To Metric
5003001 - Velocity - Racing-scale - From Metric
5004000 - Velocity - Air-scale - To Metric
5004001 - Velocity - Air-scale - From Metric
5005000 - Velocity - Rocket-scale - To Metric
5005001 - Velocity - Rocket-scale - From Metric
6001000 - Area - Home-scale - To Metric
6001001 - Area - Home-scale - From Metric
6002000 - Area - Commercial-scale - To Metric
6002001 - Area - Commercial-scale - From Metric
6003000 - Area - Land-scale - To Metric
6003001 - Area - Land-scale - From Metric
6004000 - Area - Jurisdictional-scale - To Metric
6004001 - Area - Jurisdictional-scale - From Metric
6005000 - Area - Territory-scale - To Metric
6005001 - Area - Territory-scale - From Metric
```

The Subjects Table:
```
100 - Length
200 - Mass
300 - Volume
400 - Temperature
500 - Velocity
600 - Area
```

The numbering is naively expansion receptive.

* Subjects are the first 3 digits.
* SubSubjects are the next 3 digits.
* Direction is the last digit.
  * To Metric: 0
  * From Metric: 1

The existing Subjects will be 100XXX, 200XXX, 300XXX, 400XXX, 500XXX, 600XXX.
The existing SubSubjects will be XXX100, XXX200, XXX300, XXX400, etc...

An addition would just be another in the series.

An insertion's number will be an even division between the existing numbers.
Example: A new subject between 200XXX Mass and 300XXX Volume will be 250XXX.
Example: A new scale between XXX100 Volume - Hand and XXX200 Volume - Human will be XXX150.

Example: New tenth SubSubject for Area would go like this:
```
6009000 - Area - Really-Big-scale - To Metric
6009001 - Area - Really-Big-scale - From Metric
6010000 - Area - Really-Really-Big-scale - To Metric
6010001 - Area - Really-Really-Big-scale - From Metric
```

Example: New tenth Subject:
```
9004000 - Light - Sun-scale - To Metric
9004001 - Light - Sun-scale - From Metric
10001000 - Electricity - Mini-scale - To Metric
10001001 - Electricity - Mini-scale - From Metric
```

Example: A new micro-scale mass SubSubject would be 50 less:
```
2000500 - Mass - Micro-scale - To Metric
2000501 - Mass - Micro-scale - From Metric
2001000 - Mass - Hand-scale - To Metric
2001001 - Mass - Hand-scale - From Metric
```

Example: A new subject between Temperature and Velocity:
```
4003000 - Temperature - Chemistry-scale - To Metric
4003001 - Temperature - Chemistry-scale - From Metric
4501000 - Electricity - Mini-scale - To Metric
4501001 - Electricity - Mini-scale - From Metric
5001000 - Velocity - Pedestrian-scale - To Metric
5001001 - Velocity - Pedestrian-scale - From Metric
```

Example (continued): A new subject between Electricity and Velocity:
```
4501000 - Electricity - Mini-scale - To Metric
4501001 - Electricity - Mini-scale - From Metric
4751000 - Light - Astronomy-scale - To Metric
4751001 - Light - Astronomy-scale - From Metric
5001000 - Velocity - Pedestrian-scale - To Metric
5001001 - Velocity - Pedestrian-scale - From Metric
```

Get it?

**Comment by TiE23** [comment]
When done make a new ticket to sort by this new field on UserDetailMasteries, MasteriesList, and ChallengeKickoffSelectorSubject components.
